src/libmongoc/tests/test-mongoc-retryable-writes.c

```diff
@@ -568,6 +568,111 @@ test_unsupported_storage_engine_error (void)
    mongoc_client_destroy (client);
    mock_rs_destroy (rs);
 }
+
+/* Tests original error is returned after encountering a
+ * WriteConcernError with a RetryableWriteError label.
+   The test requires a 6.0+ replica set */
+typedef struct {
+   mongoc_client_t *client;
+   bool configure_second_fail;
+} prose_test_3_apm_ctx_t;
+
+static void
+prose_test_3_command_succeeded (const mongoc_apm_command_succeeded_t *event)
+{
+   bson_iter_t iter;
+   bson_error_t error;
+   const bson_t *reply = mongoc_apm_command_succeeded_get_reply (event);
+   prose_test_3_apm_ctx_t *ctx =
+      mongoc_apm_command_succeeded_get_context (event);
+   mongoc_client_t *client = ctx->client;
+
+   // wait for a writeConcernError and then set a second failpoint
+   if (bson_iter_init_find (&iter, reply, "writeConcernError") &&
+       ctx->configure_second_fail) {
+      ctx->configure_second_fail = false;
+      ASSERT_OR_PRINT (
+         mongoc_client_command_simple (
+            client,
+            "admin",
+            tmp_bson (
+               "{'configureFailPoint': 'failCommand', 'mode': {'times': 1},"
+               " 'data': { 'failCommands': ['insert'], 'errorCode': 10107, "
+               "'errorLabels': ['RetryableWriteError', 'NoWritesPerformed']}}"),
+            NULL,
+            NULL,
+            &error),
+         error);
+   }
+}
+
+static void
+retryable_writes_prose_test_3 (void *ctx)
+{
+   mongoc_client_t *client;
+   mongoc_collection_t *coll;
+   bool ret;
+   bson_t reply;
+   bson_error_t error;
+   mongoc_uri_t *uri;
+   mongoc_apm_callbacks_t *callbacks;
+   prose_test_3_apm_ctx_t apm_ctx = {0};
+
+   int write_concern_error_code = 91;
+   BSON_UNUSED (ctx);
+
+   // setting up the client
+   uri = test_framework_get_uri ();
+   mongoc_uri_set_option_as_bool (uri, "retryWrites", true);
+   client = test_framework_client_new_from_uri (uri, NULL);
+   test_framework_set_ssl_opts (client);
+   coll = get_test_collection (client, "retryable_writes");
+
+   mongoc_uri_destroy (uri);
```

Comment:
```suggestion
   client = test_framework_new_default_client ();
   coll = get_test_collection (client, "retryable_writes");
```

Retryable writes are [enabled by default](https://github.com/mongodb/mongo-c-driver/blob/5c049bf8c0d21388fcfcd29c7606b8526e681a6a/src/libmongoc/src/mongoc/mongoc-cmd-private.h#L43). Creating a separate URI is not necessary.